libs/remote_storage/src/azure_blob.rs

```diff
@@ -152,39 +155,43 @@ impl AzureBlobStorage {
                 Err(_elapsed) => Err(DownloadError::Timeout),
             });
 
-            let mut response = std::pin::pin!(response);
-
-            let mut bufs = Vec::new();
-            while let Some(part) = response.next().await {
-                let part = part?;
-                if etag.is_none() {
-                    etag = Some(part.blob.properties.etag);
-                }
-                if last_modified.is_none() {
-                    last_modified = Some(part.blob.properties.last_modified.into());
-                }
-                if let Some(blob_meta) = part.blob.metadata {
-                    metadata.extend(blob_meta.iter().map(|(k, v)| (k.to_owned(), v.to_owned())));
-                }
-                let data = part
-                    .data
-                    .collect()
-                    .await
-                    .map_err(|e| DownloadError::Other(e.into()))?;
-                bufs.push(data);
-            }
+            let mut response = Box::pin(response);
```

Comment:
Aside, do you know what is the amount the blob storage chunks to? If you know, or even better, have a link to their docs, it would be good to include it here.

Reply:
I don't know and I don't think it's relevant. IMO it's an implementation detail like MTU or ip of the endpoint.

Reply:
Surely it cannot be the MTU.

Reply:
Oki, maybe we should consider adding it as a metric if we don't know.

Reply:
Maybe it can be a followup, idk if it's worth to add as a metric though.